woocommerce/woocommerce-android
# Replace mocked requests with real requests

Closes: #10722 


Relies on these FluxC changes to be merged first: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2977

### Description

Replaces mocked requests with real requests


### Testing instructions

1. Trigger Blaze campaign creation flow
2. Create a campaign for a product
3. Select an existing payment method from the ones fetched. Note that "Add Payment" method is not ready yet
4. ~~Check that the campaign is created successfully~~ Campaign creation will fail for now. So expect to see the error screen and an API response such this: 
```
{
   "message":"Validation Failed",
   "details":{
      "body":{
         "message":"\"budget\" is an excess property and therefore is not allowed",
         "value":{
            "budget":{
               "mode":"total",
               "amount":"5",
               "currency":"USD"
            }
         }
      },
      "body.main_image.crop":{
         "message":"'crop' is required"
      },
      "body.currency":{
         "message":"'currency' is required"
      },
      "body.total_budget":{
         "message":"'total_budget' is required"
      }
   }
}
```

We'll test everything properly as soon as the API is ready. Full context: p1710946289951609/1710926665.933569-slack-C03L1NF1EA3
